src/components/SidebarState.vue

```diff
@@ -34,72 +80,125 @@ export default Vue.extend({
-            this.sidebarUser = user;
-            this.sidebarOpen = true;
-            this.sidebarSection = 'user';
+
+            // Only set sidebar section if its in the allowed list
+            if (this.sidebarState.allowedSections.includes(section)) {
+                if (section === 'user') {
+                    let user = this.getUserFromOpts(opts);
+                    if (user) {
+                        this.sidebarUser = user;
+                        this.sidebarSection = section;
+                        this.sidebarOpen = true;
+                    }
+                    // Could be invalid user for which we do nothing
+                    return;
+                }
+                this.sidebarSection = section;
+                this.sidebarOpen = true;
+            }
+        },
+        toggleSidebar(section, opts) {
+            if (!section && this.sidebarOpen) {
+                this.close();
+                return;
+            }
+            if (section === 'user') {
+                let user = this.getUserFromOpts(opts);
+                this.sidebarSection === section && (!user || user === this.sidebarUser) ?
+                    this.close() :
+                    this.showSidebar(section, opts);
+                return;
+            }
+            this.sidebarSection === section ?
+                this.close() :
+                this.showSidebar(section, opts);
+        },
+        showUser(user) {
+            this.showSidebar('user', { user: user });
         },
         showNicklist() {
-            this.activeComponent = null;
-            this.sidebarOpen = true;
-            this.sidebarSection = 'nicklist';
+            this.showSidebar('nicklist');
         },
         showBufferSettings() {
-            this.activeComponent = null;
-            this.sidebarOpen = true;
-            this.sidebarSection = 'settings';
+            this.showSidebar('settings');
         },
         showAbout() {
-            this.activeComponent = null;
-            this.sidebarOpen = true;
-            this.sidebarSection = 'about';
+            this.showSidebar('about');
+        },
+        toggleUser(nick) {
+            let user = this.$state.getUser(this.$state.getActiveNetwork().id, nick);
+            this.toggleSidebar('user', { user: user });
         },
         toggleNicklist() {
-            this.sidebarSection === 'nicklist' ?
-                this.close() :
-                this.showNicklist();
+            this.toggleSidebar('nicklist');
         },
         toggleBufferSettings() {
-            this.sidebarSection === 'settings' ?
-                this.close() :
-                this.showBufferSettings();
+            this.toggleSidebar('settings');
         },
         toggleAbout() {
-            this.sidebarSection === 'about' ?
-                this.close() :
-                this.showAbout();
+            this.toggleSidebar('about');
+        },
+        getUserFromOpts(opts) {
```

Comment:
What's the point of this function? It reads the `opts.user` and then looks it up again from state? We pass the user object already to sidebar functions.